### PR TITLE
feat: separate VPN configurations and ccd folders

### PR DESCRIPTION
**Pull request by lemeurherve**
Follow-up of https://github.com/jenkins-infra/azure-net/pull/10

The goal of this PR is to be able to use/generate config for several VPNs.

Ref: https://github.com/jenkins-infra/helpdesk/issues/2844

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -16,7 +16,7 @@ RUN addgroup --gid 101 openvpn \
 
 COPY --chown=openvpn cert/pki/ca.crt /etc/openvpn/server/ca.crt
 COPY --chown=openvpn cert/pki/crl.pem /etc/openvpn/server/crl.pem
-COPY --chown=openvpn cert/ccd /etc/openvpn/server/ccd
+COPY --chown=openvpn cert/ccd /home/openvpn/available-ccd-folders
```
Comment (dduportal) [inline Dockerfile:19]:
```suggestion
COPY --chown=openvpn cert/ccd /home/openvpn/available-ccds
```

Nit: I'm not sure how to properly explain it but I have a feeling that having the word `folder` inside a folder name is confusing. WDYT with the plural form?

File: docker/docker-compose.yaml [hunk 2]
```diff
@@ -13,8 +13,9 @@ services:
         - "AUTH_LDAP_URL=ldaps://ldap"
         - "AUTH_LDAP_BINDDN=cn=admin,dc=jenkins-ci,dc=org"
         - "AUTH_LDAP_GROUPS_MEMBER=cn=all"
+        - "OPENVPN_NETWORK=default"
```
Comment (dduportal) [inline docker/docker-compose.yaml:16]:
```suggestion
        - "OPENVPN_NETWORK=mocked-vnet"
```

File: docker/docker-compose.yaml [hunk 3]
```diff
@@ -27,7 +28,7 @@ services:
 
   # Don't forget to run /entrypoint/restore, once database is started
   ldap:
-    image: jenkinsciinfra/ldap@sha256:bee8e769077e82984ada23d25b06c7f42130a4b9f9fa92ead20916e01c1af8de
+    image: jenkinsciinfra/ldap@sha256:185af88df34f0db27d8351e4844e9b7c04addd33917a26c2a2a5143f11fba9fa
```
[the remark this answers is not in the document]
Comment (lemeurherve) [reply docker/docker-compose.yaml:31]: LGTM. I hesitated to add an updatecli manifest, but 🤷

File: docker/entrypoint.sh [hunk 4]
```diff
@@ -44,6 +44,15 @@ function ensure_required_variables {
   : "${AUTH_LDAP_GROUPS_MEMBER:? AUTH_LDAP_GROUPS_MEMBER required}"
 }
 
+function copy_clients_certificates {
```
Comment (dduportal) [inline docker/entrypoint.sh:47]:
```suggestion
function copy_client_configurations_directory {
```

Nit: CCD stands for "Client Configuration Directory"

File: docker/entrypoint.sh [hunk 5]
```diff
@@ -44,6 +44,15 @@ function ensure_required_variables {
...
+  if [[ -z "${OPENVPN_NETWORK}" ]]; then
+    echo "No OPENVPN_NETWORK env var set, no cdd copied."
```
Comment (dduportal) [inline docker/entrypoint.sh:49]:
```suggestion
    echo "No OPENVPN_NETWORK env var set: no ccd copied."
```

File: docker/entrypoint.sh [hunk 6]
```diff
@@ -44,6 +44,15 @@ function ensure_required_variables {
...
+  else
+    mkdir -p /etc/openvpn/server/cdd
+    cp /home/openvpn/available-ccd-folders/${OPENVPN_NETWORK}/* /etc/openvpn/server/cdd
+  fi
+}
+
```
Comment (dduportal) [inline docker/entrypoint.sh:55]:
```suggestion
  else
    mkdir -p /etc/openvpn/server/ccd
    cp /home/openvpn/available-ccd-folders/${OPENVPN_NETWORK}/* /etc/openvpn/server/ccd
  fi
}

```

File: docker/entrypoint.sh [hunk 7]
```diff
@@ -59,6 +68,7 @@ function start_openvpn {
 }
 
 ensure_required_variables
+copy_clients_certificates
```
Comment (lemeurherve) [inline docker/entrypoint.sh:71]:
```suggestion
copy_client_configurations_directory
```

Comment (dduportal) [inline docker/entrypoint.sh]:
```suggestion
copy_client_configurations_directory
```

**Review by dduportal — COMMENTED** [review]
Nice work. A few nitpicks and a set of blocking typos (`s/cdd/ccd/g`)

File: docker/entrypoint.sh [hunk 8]
```diff
@@ -44,6 +44,15 @@ function ensure_required_variables {
   : "${AUTH_LDAP_GROUPS_MEMBER:? AUTH_LDAP_GROUPS_MEMBER required}"
 }
 
+function copy_client_configurations_directory {
+  if [[ -z "${OPENVPN_NETWORK}" ]]; then
+    echo "No OPENVPN_NETWORK env var set: no ccd copied."
+  else
+    mkdir -p /etc/openvpn/server/cdd
+    cp /home/openvpn/available-ccd-folders/${OPENVPN_NETWORK}/* /etc/openvpn/server/cdd
```
Comment (lemeurherve) [inline docker/entrypoint.sh:52]:
```suggestion
    mkdir -p /etc/openvpn/server/ccd
    cp /home/openvpn/available-ccds/${OPENVPN_NETWORK}/* /etc/openvpn/server/ccd
```

File: docker/docker-compose.yaml [hunk 9]
```diff
@@ -13,8 +13,9 @@ services:
         - "AUTH_LDAP_URL=ldaps://ldap"
         - "AUTH_LDAP_BINDDN=cn=admin,dc=jenkins-ci,dc=org"
         - "AUTH_LDAP_GROUPS_MEMBER=cn=all"
+        - "OPENVPN_NETWORK=mocked-vnet"
     volumes:
-        - ./mock/vpn/ccd:/etc/openvpn/server/ccd:ro
+        - ./mock/vpn/ccd:/home/openvpn/available-ccd-folders:ro
```
Comment (lemeurherve) [inline docker/docker-compose.yaml:18]:
```suggestion
        - ./mock/vpn/ccd:/home/openvpn/available-ccds:ro
```

**Comment by timja** [comment]
Is there any documentation updates required as part of this?

File: utils/easyvpn/cmd/revoke.go [hunk 10]
```diff
@@ -17,6 +17,7 @@ func init() {
 	revokeCmd.Flags().BoolVarP(&Commit, "commit", "", true, "git commit changes")
 	revokeCmd.Flags().BoolVarP(&Push, "push", "", true, "git push changes")
 	revokeCmd.Flags().StringVarP(&CertDir, "cert", "c", "cert", "Cert Directory")
+	revokeCmd.Flags().StringVarP(&Network, "network", "", "default", "Network")
```
Comment (dduportal) [inline utils/easyvpn/cmd/revoke.go:20]:
```suggestion
	revokeCmd.Flags().StringVarP(&Network, "network", "n", "default", "Network")
```

Is there a particular reason to not provide the short flag form?

**Comment by lemeurherve** [comment]
Yes, I shouldn't have put this PR in "ready for review", switching to draft again.

Essentially the changes will concern admin instructions, so the ccd are stored in "default" and/or "private"

File: utils/easyvpn/cmd/revoke.go (hunk 10 above)
Comment (lemeurherve) [reply utils/easyvpn/cmd/revoke.go:20]: Yes, to be coherent with the flag definition already present in sign.go

I can add the short flag on both though.

Comment (dduportal) [reply utils/easyvpn/cmd/revoke.go:20]: yep, that would help to have both long and short flag  for the 2 commands

**Comment by dduportal** [comment]
> Is there any documentation updates required as part of this?

Once it will be finished: yes \o/ (but now, no effort on the documentation as this is a set of moving parts)

**Review by dduportal — APPROVED** [review]

**Pull request merged**